src/charts/scale.js

```diff
@@ -55,12 +55,16 @@ export default function Scale () {
   function buildXScale (_allKeys) {
     let xScale = null
     let domain = null
-    const markW =  hasBars(config.chartType) ? cache.chartWidth / _allKeys.length : 0
+    const isBarChart = hasBars(config.chartType)
+    const markW = 0 //  hasBars(config.chartType) ? cache.chartWidth / _allKeys.length : 0
 
     if (config.keyType === "time") {
       xScale = d3.scaleTime()
     } else if (config.keyType === "number") {
       xScale = d3.scaleLinear()
+    } else if (config.keyType === "string" && isBarChart) {
```

Comment:
I need to test how this logic handle the case where chartType is for example `["bar", "bar", "bar"]` (as can be the case in combo chart) instead of type `"bar"`. Also, this will probably break when using `stackedBar` with time and number keyTypes, which is not a requirement in Immerse right now, but is already available in mapd3 and may become a requirement when we will replace the histogram for example.

Reply:
Makes sense to think ahead on this one. The more "d3 way" of doing it seems to be using two ordinal scales, one for the grouping and one for the group category: https://bl.ocks.org/mbostock/3887051

Benefit seems to be that it makes adjusting the padding between bars and groups more declarative. It might also be more familiar to other d3 devs who come across this code.

Reply:
I agree. But we do have numerical and time axis on group bar chart (combo) already, so that may break.

Reply:
I'm fairly certain that you can treat numeric and temporal values as ordinal with d3.scaleBand, but would need to double check.